swift/mt940.go

```diff
@@ -235,13 +236,13 @@ func (t *TransactionTag) Unmarshal(value []byte) error {
 	if unicode.IsDigit(r) {
 		buf.UnreadRune()
 		dateBytes = buf.Next(4)
-		date, err = parseDate(dateBytes, t.ValutaDate.Year())
+		date, err = parseDate(dateBytes, bookingYear)
 		if err != nil {
 			return errors.WithMessage(err, "unmarshal transaction tag: parsing booking date")
 		}
 		t.BookingDate = domain.NewShortDate(date)
-		monthDiff := int(math.Abs(float64(t.ValutaDate.Month() - t.BookingDate.Month())))
-		if monthDiff > 1 {
+		diff := t.ValutaDate.Sub(t.BookingDate.Time).Hours() / 24
+		if diff > 31 {
```

Comment:
Unfortunately I haven't yet understood what we're calculating here and why we're checking for 31.

Reply:
how may days can be between a booking and a valuta date? I am sure it us less then 31 days.

Reply:
regarding this thread https://homebanking-hilfe.de/forum/topic.php?t=8970 which may or may not be representative the number of days can be way higher